kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/collections.kt

```diff
@@ -33,11 +63,55 @@ fun <A> Arb.Companion.set(gen: Gen<A>, range: IntRange = 0..100): Arb<Set<A>> =
  * Shrinking is performed by removing elements from the list until the empty list.
  */
 @JvmOverloads
-fun <A> Arb.Companion.list(gen: Gen<A>, range: IntRange = 0..100): Arb<List<A>> = arb(ListShrinker()) { rs ->
-   val size = rs.random.nextInt(range)
-   gen.generate(rs).take(size).map { it.value }.toList()
+fun <A> Arb.Companion.list(gen: Gen<A>, range: IntRange = 0..100): Arb<List<A>> = object : Arb<List<A>> {
```

Comment:
I changed `list` and `set` because the provided gen may be an arb with edgecases. In that case, something like this

```kotlin
forAll(100, Arb.list(myArbWithEdgeCases()) { testList -> 
   ...
}
```

would result in every provided testList starting with the edgecases.